static/js/services/contact-view-model-generator.js

```diff
@@ -147,7 +147,25 @@ angular.module('inboxServices').factory('ContactViewModelGenerator',
       return DB().query('medic-client/contacts_by_parent', {
         key: contactId,
         include_docs: true
-      });
+      })
+        .then(function(childrenResponse) {
+          var ids = childrenResponse.rows.map(function(child) {
+            return child.doc.contact && child.doc.contact._id;
+          });
+          return DB().query('medic-client/doc_summaries_by_id', { keys: ids })
+            .then(function(contactsResponse) {
+              childrenResponse.rows.forEach(function(row) {
+                var contactId = row.doc.contact && row.doc.contact._id;
+                if (contactId) {
+                  var contactRow = _.findWhere(contactsResponse.rows, { id: contactId });
+                  if (contactRow) {
+                    row.doc.contact.name = contactRow.value.name;
+                  }
+                }
+              });
+              return childrenResponse;
```

Comment:
I find it weird to return the full db object here, can you return just the children array? 

Edit : ok, looks like the subsequent code and the view expect a `rows` obj, and calls `.doc` everywhere, which I find weird too. Can we get rid of that, and just pass around the docs themselves? (if you find that out of scope, then at least return the `rows` from this function, rather than the whole db object)